src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size);
+        if (isEmpty()) {
+            addFirst(value);
+        } else if (index == size) {
+            addLast(value);
+        } else if (index != size) {
+            addByIndex(value, index);
+        }
+    }
+
+    private void checkIndex(int index, int limit) {
+        if (index > limit || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+    }
+
+    private void addByIndex(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> tempNode = findNodeByIndex(index);
+        if (tempNode.equals(first)) {
+            first = newNode;
+        } else {
+            tempNode.prev.next = newNode;
+        }
+        newNode.next = tempNode;
+        newNode.prev = tempNode.prev;
+        tempNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+    }
+
+    private void addFirst(T value) {
+        first = new Node(null, value, null);
+        last = first;
+        size++;
+    }
+
+    private void addLast(T value) {
+        Node<T> tempNode = new Node<T>(last, value, null);
+        last.next = tempNode;
+        last = tempNode;
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeByIndex(index).value;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        checkIndex(index, size - 1);
+        Node<T> result = first;
+        for (int i = 0; i < index; i++) {
+            result = result.next;
+        }
+        return result;
```

Comment:
Check the index. If it is more than size / 2, then it is better to move towards it from the other side.